packages/gatsby/cache-dir/json-store.js

```diff
@@ -72,6 +72,12 @@ class JSONStore extends React.Component {
     // - page data for path changed
     // - static query results changed
 
+    // Always return false if we're missing resources.
+    const data = nextState.pageQueryData[normalizePagePath(nextState.path)]
+    if (!data) {
+      return false
```

Comment:
With this approach, e2e tests, which wait for `onRouteUpdate` and check focused element, randomly fail, because of the route changed but page-data did not load yet.

Reply:
Or maybe instead of this, change render function like this:

```js
  render() {
    const data = this.state.pageQueryData[getPathFromProps(this.props)]
	let { pageResources, ...otherProps } = this.props
    // eslint-disable-next-line
    if (!data) {
	  pageResources = Object.assign({}, pageResources, { component: () => <div /> })
    }
    return (
      <StaticQueryContext.Provider value={this.state.staticQueryData}>
        <PageRenderer pageResources={pageResources} {...otherProps} {...data} />
      </StaticQueryContext.Provider>
    )
  }
```